src/batMap.ml

```diff
@@ -1470,6 +1729,20 @@ module PMap = struct (*$< PMap *)
   let merge_unsafe f m1 m2 =
     { m1 with map = Concrete.merge f m1.cmp m1.map m2.map }
 
+  let of_seq ?(cmp = Pervasives.compare) s =
+    { map = Concrete.of_seq cmp s; cmp = cmp }
+    
+  let to_seq m = Concrete.to_seq m.map
+             
+  let to_seq_from k m =
+    Concrete.to_seq_from m.cmp k m.map
+                      
+  let add_seq s m =
+    { m with map = Concrete.add_seq m.cmp s m.map }
+
+  let union_stdlib f m1 m2 =
+    { m1 with map = Concrete.union_stdlib m1.cmp f m1.map m2.map }
```

Comment:
This implementation is wrong if `m1.cmp` is not compatible with `m2.map`. See how `heuristic_merge` is implemented.

Reply:
avoided by implementing union_stdlib in terms of heuristic_merge instead